### PR TITLE
refactor(makefile): add PCRE_INCDIR variable

### LINKED ISSUE #8432: bug: for mac making deps,pcre header file cannot be found
### Current Behavior

after executing `make deps` on my mac m1 chip version,terminal showed the result below.
image

And I have installed the pcre by homebrew before `make deps`.
image

### Expected Behavior

luarocks can find the header file `pcre.h`.

### Error Logs

_No response_

### Steps to Reproduce

1. Run `make deps`

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`): Darwin raymondMacBook-Pro.local 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000 arm64
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by drgnchan**
### Description

Fixes #8432

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -153,6 +156,7 @@ deps: runtime
 		mkdir -p ~/.luarocks; \
 		$(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.OPENSSL_LIBDIR $(addprefix $(ENV_OPENSSL_PREFIX), /lib); \
 		$(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.OPENSSL_INCDIR $(addprefix $(ENV_OPENSSL_PREFIX), /include); \
+		$(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.PCRE_INCDIR $(addprefix $(ENV_PCRE_PREFIX), /include); \
```
Comment (tokers) [inline Makefile:159]: The variable `ENV_PCRE_PREFIX` will only be set in darwin, so this command should be conditional.

Comment (drgnchan) [reply Makefile:159]: Thank you for the reminder.Actually I found `pcre.h` in `/usr/include` in my Ubuntu vm.I will wrap this line by an `ifeq` condition for darwin.What do you think of the condition i am going to add?Is it enough?

**Comment by drgnchan** [comment]
I have tested with Linux and Darwin:
1. `rm -rf deps`
2. `rm ~/.luarocks/config-5.1.lua`
3. rerun `make deps`

Finally both linux and darwin install jsonschema(requeir `pcre`) successfully.

File: Makefile (hunk 1 above)
Comment (tokers) [reply Makefile]: Just check if this variable exists.

**Comment by drgnchan** [comment]
I am not sure why  the CI job [build (ubuntu-20.04, linux_openresty_1_19, t/plugin/[a-k]*)](https://github.com/apache/apisix/actions/runs/3584113848/jobs/6039949287#logs) succeeded but [build (ubuntu-20.04, linux_openresty, t/plugin/[a-k]*)](https://github.com/apache/apisix/actions/runs/3584113848/jobs/6039948982#logs) failed.And I think the difference of the two job is the openresty version.

**Comment by tzssangglass** [comment]
the failed CI `CI Centos7 / run ci on centos7 (t/node t/pubsub t/router t/script t/stream-node t/utils t/wasm t/xds-library) (pull_request) ` has been fixed by #8438, pls merge upstream master branch to your PR.

**Comment by tzssangglass** [comment]
hi @drgnchan I think you are mistaken, your PR has other people's commits in it.

**Comment by drgnchan** [comment]
@tzssangglass I am sorry having made the wrong workflow before.I just sync fork by Github and I think it is ok now.

File: Makefile [hunk 2]
```diff
@@ -153,6 +156,7 @@ deps: runtime
 		mkdir -p ~/.luarocks; \
 		$(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.OPENSSL_LIBDIR $(addprefix $(ENV_OPENSSL_PREFIX), /lib); \
 		$(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.OPENSSL_INCDIR $(addprefix $(ENV_OPENSSL_PREFIX), /include); \
+		[ '$(ENV_OS_NAME)' == 'darwin' ] && $(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.PCRE_INCDIR $(addprefix $(ENV_PCRE_PREFIX), /include); \
```
Comment (tzssangglass) [inline Makefile:159]: without `variables. PCRE_DIR` ?

Comment (drgnchan) [reply Makefile:159]: image
Yes,pcre suggests PCRE_DIR or PCRE_INCDIR.I will add PCRE_DIR.

Comment (tzssangglass) [reply Makefile:159]: try with 

```
[ '$(ENV_OS_NAME)' == 'darwin' ] && $(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.PCRE_DIR $(addprefix $(ENV_PCRE_PREFIX), /lib); \
```

Comment (tzssangglass) [reply Makefile:159]: form the error message https://github.com/apache/apisix/issues/4945

```
You may have to install PCRE in your system and/or pass PCRE_DIR or PCRE_INCDIR to the luarocks command.
```

sorry, forget what I said, it looks like we only need one of PCRE_DIR or PCRE_INCDIR. 

I test this PR on macOS(amd64), and it works well. 

so we can remove 

```
[ '$(ENV_OS_NAME)' == 'darwin' ] && $(ENV_LUAROCKS) config $(ENV_LUAROCKS_FLAG_LOCAL) variables.PCRE_LIBDIR $(addprefix $(ENV_PCRE_PREFIX), /lib); \
```

BTW, I need to add 

```
   LDAP_DIR = "/usr/local/opt/openldap",
   LDAP_INCDIR = "/usr/local/opt/openldap/include",
```

to `~/.luarocks/config-5.1.lua`

Comment (drgnchan) [reply Makefile:159]: All right.I have removed it.

**Review by leslie-tsang — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Comment by Sn0rt** [comment]
LGTM
 
the branch has been tested at a M2 machine.

**Comment by monkeyDluffy6017** [comment]
@Sn0rt Thanks

**Pull request merged**